### DISCUSSION TITLE
Is there a way to use this with uvu?

### DISCUSSION THREAD

**Discussion by jasikpark** [dubzzz/fast-check#2004] [Q&A] [answered]
astro uses https://github.com/lukeed/uvu for it's testing framework, and I was wondering if it fast-check would be useable for specifying race-condition tests?

**Comment by dubzzz** [comment, marked as answer]
Normally fast-check is supposed to be framework agnostic so yes.

Actually from what I see in the documentation of `uvu`, you will have to wrap the calls to fast-check within the `test` function as follow:
```js
// For synchronous properties
test('My property', () => {
  fc.assert(fc.property(
    // here your arbitraries
    // here the test callback
  ));
});

// For asynchronous properties
test('My async property', async () => {
  await fc.assert(fc.asyncProperty(
    // here your arbitraries
    // here the test callback
  ));
});
// Alternatively if you return the output of fc.assert (just hiding the async/await but uvu still sees Promises)
test('My async property', () => 
  fc.assert(fc.asyncProperty(
    // here your arbitraries
    // here the test callback
  ))
);
```

In order to use the scheduler you may have something like:
```js
test('My counter', async () => {
  await fc.assert(fc.asyncProperty(fc.scheduler(), async (s) => {
    // Arrange
    let dbValue = 0;
    const db = {
      read: s.scheduleFunction(async function read() {
        return dbValue;
      }),
      write: s.scheduleFunction(async function write(newValue: number, oldValue?: number) {
        if (oldValue !== undefined && dbValue !== oldValue) return false;
        dbValue = newValue;
        return true;
      }),
    };
    const counter = new Counter(db);

    // Act
    s.schedule(Promise.resolve('inc1')).then(() => counter.inc());
    s.schedule(Promise.resolve('inc2')).then(() => counter.inc());
    await s.waitAll();

    // Assert
    expect(dbValue).toBe(2);
  }));
});
```
_Inspired by https://codesandbox.io/s/github/dubzzz/fast-check/tree/main/example?previewwindow=tests&file=/005-race/counter/main.spec.ts_

**Reply by jasikpark** [reply]
thanks! sorry that I never saw this - I'm now trying out fast-check at my job

currently trying to figure out where it would fit in - I set it up for a utility function, but for tests of react storybooks, where normally we'd interact with a mocked component and then query the state of the dom, I'm unsure how it'd fit together
